content/academy/apify_platform/deploying/inputs_outputs.md

```diff
@@ -0,0 +1,193 @@
+
+### Accepting input with the Apify SDK
+
+Since we're using Node.js, let's install the **apify** package by running the following command:
+
+```shell
+npm install apify
+```
+
+Now, let's import `Apify` and use the `Apify.getInput()` function to grab our input.
+
+```JavaScript
+// index.js
+import Apify from 'apify';
+
+const input = await Apify.getInput();
+console.log(input);
+```
+
+If we run this right now, we'll see **null** in our terminal - this is because we never provided any sort of test input, which should be provided in the default key-value store. The `Apify.getInput()` function has detected that there is no **apify_storage** folder and generated one for us. We'll now add an **INPUT.json** file within **apify_storage/key_value_stores/default** to match what we're expecting in our code.
+
+```JSON
+{
+    "numbers": [5, 5, 5, 5]
+}
+```
+
+Then we can modify our code to grab the input and use it to generate a solution which is logged to the console.
+
+```JavaScript
+// index.js
+import Apify from 'apify';
+
+const { numbers } = await Apify.getInput();
+
+const addAllNumbers = (...nums) => nums.reduce((total, curr) => (total += curr));
+
+const solution = addAllNumbers(...numbers);
+
+console.log(solution);
+```
+
+Cool! When we run `node index.js`, we see **20**.
+
+### Accepting input without the Apify SDK
+
+Alternatively, when writing in a language other than JavaScript, we can create our own `get_input()` function which utilizes the Apify API. For this example, we are using the [Apify Client]({{@link apify_platform/getting_started/apify_client.md}}) for Python to access the API.
+
+```Python
+# index.py
+from apify_client import ApifyClient
+from os import environ
+import json
+
+client = ApifyClient(token='YOUR_TOKEN')
+
+# If being run on the platform, the "APIFY_IS_AT_HOME" environment variable
+# will be "1". Otherwise, it will be undefined/None
+def is_on_apify ():
+    if (environ.get('APIFY_IS_AT_HOME')):
+        return True
+    return False
+
+# Get the input
+def get_input ():
+    if (not is_on_apify()):
+        input = open('./apify_storage/key_value_stores/default/INPUT.json')
+        return json.load(input)
+
+    kv_store = client.key_value_store(environ.get('APIFY_DEFAULT_KEY_VALUE_STORE_ID'))
+    kv_store.get_record('INPUT', format='json')
+
+def add_all_numbers (nums):
+    total = 0
+
+    for num in nums:
+        total += num
+
+    return total
+
+solution = add_all_numbers(get_input()['numbers'])
+
+print(solution)
+```
+
+> For a better understanding of API endpoints reading and modifying key-value stores, check the [official API reference](https://docs.apify.com/api/v2#/reference/key-value-stores).
+
+## [](#writing-output) Writing output
+
+Similarly to reading input, you can write the actor's output either by using the Apify SDK in Node.js, or by manually writing a utility function to do so.
```

Comment:
I'm wondering if this would be a good opportunity to quickly explain datasets and key-value stores and link to their docs. Without that, the output saving is a bit magical and begs the question where does it actually save.